src/cache/s3.rs

```diff
@@ -29,6 +31,11 @@ impl S3Cache {
         server_side_encryption: Option<bool>,
     ) -> Result<Operator> {
         let mut builder = S3::default();
+        let user_agent = format!("{}/{}", env!("CARGO_PKG_NAME"), env!("CARGO_PKG_VERSION"));
+        let client_builder = ClientBuilder::new().user_agent(user_agent);
+        let client =
+            HttpClient::build(client_builder).map_err(|err| err.with_operation("S3::build"))?;
+        builder.http_client(client);
```

Comment:
what is going to be the user agent ?

Reply:
the user agent is a string whose value is `sccache/<CARGO_PKG_VERSION>` (e.g `sccache/0.7.7`).

is that what you're asking?

Reply:
yeah :)

I wonder if we could add a test for this
@Xuanwo

Reply:
> I wonder if we could add a test for this

I'm guessing we don't need extra tests, the existing s3 test covers this.